platform/ios/src/NSOrthography+MGLAdditions.m

```diff
@@ -0,0 +1,33 @@
+#import "NSOrthography+MGLAdditions.h"
+
+@implementation NSOrthography (MGLAdditions)
+
++ (NSString *)mgl_dominantScriptForMapboxStreetsLanguage:(NSString *)language {
+    if (@available(iOS 11.0, *)) {
+        NSLocale *locale = [NSLocale localeWithLocaleIdentifier:language];
+        NSOrthography *orthography = [NSOrthography defaultOrthographyForLanguage:locale.localeIdentifier];
+
+        return orthography.dominantScript;
+    }
+
+    // Manually map Mapbox Streets languages to ISO 15924 script codes.
+    NSSet *latinLanguages = [NSSet setWithObjects:@"de", @"en", @"es", @"fr", @"pt", nil];
+    NSSet *hansLanaguages = [NSSet setWithObjects:@"zh", @"zh-Hans", nil];
+    NSSet *cyrillicLanguages = [NSSet setWithObjects:@"ru", nil];
+    NSSet *arabicLanguages = [NSSet setWithObjects:@"ar", nil];
+
+    if ([latinLanguages containsObject:language]) {
+        return @"Latn";
+    } else if ([hansLanaguages containsObject:language]) {
+        return @"Hans";
```

Comment:
It would be more correct to say that `zh` is written in `Hani`, not necessarily `Hans`.

Reply:
`+[NSOrthography defaultOrthographyForLanguage:]` says that `zh = Hans`. 🤔

Reply:
It’s complicated. 🤷‍♂️